src/NHibernate/Loader/Criteria/CriteriaQueryTranslator.cs

```diff
@@ -863,6 +863,20 @@ public string GetEntityName(ICriteria subcriteria, string propertyName)
 			return GetEntityName(subcriteria);
 		}
 
+		internal IDictionary<string, string> GetCriteriaSQLAliasMap()
+		{
+			var result = criteriaSQLAliasMap.Where(p => !string.IsNullOrEmpty(p.Key.Alias)).ToDictionary(p => p.Key.Alias, p => p.Value);
+			if (outerQueryTranslator != null)
+			{
+				foreach (var p in outerQueryTranslator.GetCriteriaSQLAliasMap())
+				{
+					result.Add(p.Key, p.Value);
+				}
+			}
```

Comment:
Or maybe something like:
```suggestion
			var result = (outerQueryTranslator.GetCriteriaSQLAliasMap() ?? criteriaSQLAliasMap.Where(p => !string.IsNullOrEmpty(p.Key.Alias))).ToDictionary(p => p.Key.Alias, p => p.Value);
			if (outerQueryTranslator != null)
			{
				foreach (var p in criteriaSQLAliasMap.Where(p => !string.IsNullOrEmpty(p.Key.Alias)))
				{
					result[p.Key] = p.Value;
				}
			}
```

Reply:
This is how I coded it initially.

Reply:
We should override the `outerQueryTranslator `'s aliases.

Reply:
I'll add test

Reply:
Done